database/connect.ts

```diff
@@ -0,0 +1,37 @@
+import postgres, { Sql } from 'postgres';
+import { setEnvironmentVariables } from '../util/config';
+
+setEnvironmentVariables();
+
+declare module globalThis {
+  let postgresSqlClient: Sql;
+}
+
+// Connect only once to the database
+// https://github.com/vercel/next.js/discussions/26427#discussioncomment-898067
+function connectOneTimeToDatabase() {
+  if (!('postgresSqlClient' in globalThis)) {
+    globalThis.postgresSqlClient = postgres({
+      transform: {
+        ...postgres.camel,
+        undefined: null,
+      },
+    });
+  }
+
+  // Use Next.js Dynamic Rendering in all database queries:
+  //
+  // Wrap sql`` tagged template function to call `noStore()` from
+  // next/cache before each database query. `noStore()` is a
+  // Next.js Dynamic Function, which causes the page to use
+  // Dynamic Rendering
+  //
+  // https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic-rendering
+  return ((
+    ...sqlParameters: Parameters<typeof globalThis.postgresSqlClient>
+  ) => {
+    return globalThis.postgresSqlClient(...sqlParameters);
+  }) as typeof globalThis.postgresSqlClient;
+}
```

Comment:
As we don't need this `noStore` in expo, we may instead connect to the database like in the GraphQL example which is simpler

https://github.com/upleveled/graphql-example-spring-2024-atvie/blob/65f8a40fe857796e7cd649eb3ca5e4c8f2db3556/database/connect.ts#L16-L24

```ts
function connectOneTimeToDatabase() {
  if (!globalThis.postgresSqlClient) {
    globalThis.postgresSqlClient = postgres(postgresConfig);
  }
  return globalThis.postgresSqlClient;
}

// Connect to PostgreSQL
export const sql = connectOneTimeToDatabase();
```

Reply:
Updated in this PR
https://github.com/upleveled/graphql-example-spring-2024-atvie/pull/1